java-runtime-builder-app/src/main/java/com/google/cloud/runtimes/builder/config/ConfigFileFinder.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.runtimes.builder.config;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Optional;
+
+public abstract class ConfigFileFinder {
```

Comment:
I think that this ConfigFileFinder class hierarchy may be adding more complexity than is necessary. 

We are already checking for the presence of a Dockerfile at the root (and failing if it's found) in the [`BuildContext` class](https://github.com/GoogleCloudPlatform/runtime-builder-java/blob/master/java-runtime-builder-app/src/main/java/com/google/cloud/runtimes/builder/config/domain/BuildContext.java#L148) . IMO the check for the file at `src/main/docker/Dockerfile` should happen alongside that.

Reply:
Yea I agree, I'll move it there.